Composer/packages/client/src/pages/language-generation/code-editor.tsx

```diff
@@ -7,51 +7,79 @@ import { LgEditor, LGOption } from '@bfc/code-editor';
 import get from 'lodash/get';
 import debounce from 'lodash/debounce';
 import isEmpty from 'lodash/isEmpty';
-import { LgFile } from '@bfc/indexers';
 import { editor } from '@bfcomposer/monaco-editor/esm/vs/editor/editor.api';
 import { lgIndexer, Diagnostic } from '@bfc/indexers';
+import { RouteComponentProps } from '@reach/router';
 
 import { StoreContext } from '../../store';
 import * as lgUtil from '../../utils/lgUtil';
 
 const { check, isValid, combineMessage } = lgIndexer;
 
-interface CodeEditorProps {
-  file: LgFile;
-  template: lgUtil.Template | null;
-  line: number;
-}
-
 const lspServerPath = '/lg-language-server';
 
-export default function CodeEditor(props: CodeEditorProps) {
-  const { actions } = useContext(StoreContext);
-  const { file, template, line } = props;
+interface CodeEditorProps extends RouteComponentProps<{}> {
+  fileId?: string;
+}
+
+const CodeEditor: React.FC<CodeEditorProps> = props => {
+  const { actions, state } = useContext(StoreContext);
+  const { lgFiles } = state;
+  const file = lgFiles.length ? lgFiles[0] : null;
   const [diagnostics, setDiagnostics] = useState(get(file, 'diagnostics', []));
   const [content, setContent] = useState('');
   const [errorMsg, setErrorMsg] = useState('');
   const [lgEditor, setLgEditor] = useState<editor.IStandaloneCodeEditor | null>(null);
 
-  const fileId = file && file.id;
+  const search = props.location ? props.location.search : '';
+  const templateMatched = /^\?t=([-\w]+)/.exec(search);
+  const templateId = templateMatched ? decodeURIComponent(templateMatched[1]) : undefined;
+  const template = templateId && file ? file.templates.find(({ name }) => name === templateId) : undefined;
+
+  const hash = props.location ? props.location.hash : '';
+  const lineMatched = /L(\d+)/g.exec(hash);
```

Comment:
`query-string`

Reply:
This package probably doesn't work with the fragment format (`Lxx`). Question: why do it that way instead of `L=xx`?

Reply:
`#L5` is learned from url fragment, and some github link like 

```https://github.com/microsoft/BotFramework-Composer/blob/7119916f67466ee70f5c48dc36c2071587c5d629/Composer/packages/client/src/pages/language-generation/code-editor.tsx#L40```

```
https://github.com/microsoft/BotFramework-Composer/pull/1756#discussion_r358477500
```
it maybe sampler than `L=xx`